driver/logcumsumexp_driver.hpp

```diff
@@ -0,0 +1,453 @@
+        miopenDestroyTensorDescriptor(doutputDesc);
+        miopenDestroyTensorDescriptor(dinputDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    bool runForwardGPU = false;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t doutputDesc;
+    miopenTensorDescriptor_t dinputDesc;
+
+    std::unique_ptr<GPUMem> input_dev;
+    std::unique_ptr<GPUMem> output_dev;
+    std::unique_ptr<GPUMem> doutput_dev;
+    std::unique_ptr<GPUMem> dinput_dev;
+
+    std::vector<Tgpu> input;
+    std::vector<Tgpu> output;
+    std::vector<Tgpu> doutput;
+    std::vector<Tgpu> dinput;
+
+    std::vector<Tref> output_host;
+    std::vector<Tref> dinput_host;
+
+    int dim;
+    bool exclusive;
+    bool reverse;
+};
+
+template <typename Tgpu, typename Tref>
+int LogCumSumExpDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+
+    auto inTensorParam = inflags.GetValueTensor("input");
+    auto input_length  = inTensorParam.lengths;
+    if(input_length.empty())
+    {
+        std::cout << "Tensor must not be empty";
+        return miopenStatusBadParm;
+    }
+
+    int contiguous = inflags.GetValueInt("Contiguous");
+    if(contiguous != 0 && contiguous != 1)
+    {
+        std::cerr << "Error Tensor Contiguous should be 0 or 1" << std::endl;
+        return miopenStatusBadParm;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int LogCumSumExpDriver<Tgpu, Tref>::GetandSetData()
+{
+    dim       = inflags.GetValueInt("dim");
+    exclusive = (inflags.GetValueInt("exclusive") != 0);
+    reverse   = (inflags.GetValueInt("reverse") != 0);
+
+    auto lengths = inflags.GetValueTensor("input").lengths;
+    auto strides = GetStrides(lengths, inflags.GetValueInt("Contiguous") != 0);
+
+    if(SetTensorNd(inputDesc, lengths, strides, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input tensor: " + inflags.GetValueStr("input") + ".");
+
+    if(SetTensorNd(outputDesc, lengths, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing output tensor" + inflags.GetValueStr("input") + ".");
+
+    if(SetTensorNd(doutputDesc, lengths, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing output gradient tensor" + inflags.GetValueStr("input") + ".");
+
+    if(SetTensorNd(dinputDesc, lengths, strides, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input gradient tensor: " + inflags.GetValueStr("input") + ".");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int LogCumSumExpDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward LogCumSumExp (Default=1)", "int");
+    inflags.AddTensorFlag("input", 'D', "256x4x256", "input tensor descriptor");
```

Comment:
It would be helpful to have an input sample for the dims

Reply:
Doesn't `256x4x256` is the default input dim. If you mean that I should give an example in the description after, then I did.
```suggestion
    inflags.AddTensorFlag("input", 'D', "256x4x256", "Input tensor descriptor (Default=256x4x256)");
```